File: src/main/java/com/synopticProject/quizManager/service/QuizService.java
package com.synopticProject.quizManager.service;
import com.synopticProject.quizManager.model.Quiz;

import java.util.List;

public interface QuizService {
    Quiz getQuiz(Long id);

    List<Quiz> getAllQuizzes();

    Long createQuiz(String title);

    Quiz updateQuiz(Long id, String title);

    void deleteQuiz(Long id);
}

File: react-frontend/src/service/QuizService.js
import axios from "axios";


const getAllQuizzes = (username, password) => {
    return axios.get("http://localhost:8080/quiz/",
        {
            auth: {
                username: username,
                password: password,
            }
        })

};

const getQuiz = (id, username, password) => {
    return axios.get(`http://localhost:8080/quiz/${id}`,
        {
            auth: {
                username: username,
                password: password,
            }
        })
}

const createQuiz = (data, username, password) => {
    return axios.post("http://localhost:8080/quiz/", data,
        {
            auth: {
                username: username,
                password: password,
            }
        })
}

const updateQuiz = (id, data, username, password) => {
    return axios.put(`http://localhost:8080/quiz/${id}`, data, {
        auth: {
            username: username,
            password: password,
        }
    })
}

const QuizService = {
    getAllQuizzes, getQuiz, createQuiz, updateQuiz
};

export default QuizService;
